### ISSUE TITLE
Fix S1854 FN: Support raw string literals

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#6300]
### Description

FN for S1854 (DeadStores).

Using C#11 raw string literals, specifically with top-level statements, the rule is not raising an issue.

### Repro steps

```csharp
string x = "";   // FN
x = """Test2"""; // FN
Foo(x);
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.

**Comment by csaba-sagi-sonarsource** [comment]
I have opened [this](https://github.com/SonarSource/sonar-dotnet/issues/6323) issue, which should fix the problem. The root cause of it is the fact that the rule does not support top-level statements. It has nothing to do with raw string literals.
